contracts/abstracts/ValidatorSetBase.sol

```diff
@@ -409,7 +403,8 @@ contract ValidatorSetBase is EternalStorage, IValidatorSet {
     // Adds `_who` to the array of inactive pools
     function _addToPoolsInactive(address _who) internal {
         address[] storage poolsInactive = addressArrayStorage[POOLS_INACTIVE];
-        if (poolsInactive[poolInactiveIndex(_who)] != _who) {
+        uint256 index = poolInactiveIndex(_who);
+        if (index + 1 > 0 && poolsInactive[index] != _who) {
```

Comment:
Would `index != ~0` work, too? Might be slightly less confusing.
We should also add to the documentation of the `…Index` methods that `~0` is returned if the entry doesn't exist.